ql/pricingengines/asian/mc_discr_arith_av_price_heston.hpp

```diff
@@ -0,0 +1,226 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file mc_discr_arith_av_price_heston.hpp
+    \brief Heston MC engine for discrete arithmetic average price Asian
+*/
+
+#ifndef quantlib_mc_discrete_arithmetic_average_price_asian_heston_engine_hpp
+#define quantlib_mc_discrete_arithmetic_average_price_asian_heston_engine_hpp
+
+#include <ql/pricingengines/asian/mcdiscreteasianenginebase.hpp>
+#include <ql/processes/hestonprocess.hpp>
+#include <ql/exercise.hpp>
+
+namespace QuantLib {
+
+    //!  Heston MC pricing engine for discrete arithmetic average price Asian
+    /*!  \ingroup asianengines
+         \test the correctness of the returned value is tested by
+               reproducing results available in literature.
+    */
+    template <class RNG = PseudoRandom,
+              class S = Statistics, class P = HestonProcess>
+    class MCDiscreteArithmeticAPHestonEngine
+        : public MCDiscreteAveragingAsianEngineBase<MultiVariate,RNG,S> {
+      public:
+        typedef
+        typename MCDiscreteAveragingAsianEngineBase<MultiVariate,RNG,S>::path_generator_type
+            path_generator_type;
```

Comment:
We used to keep sources to 80 chars or so in the old days, but maybe these typedefs might be more readable on one line? I don't have strong feelings about this, though.

Reply:
Strongly agree. I was trying to follow the old style before, but definitely prefer them on the same line - will push a change now.